test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+        let(:items) {
+          %w(desktop server).map { |i| { "value" => i, "label" => i.capitalize } }
+        }
+
+        it "creates a ComboBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expected_options = [
+            client.Item(client.Id("desktop"), "Desktop", true),
+            client.Item(client.Id("server"), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], expected_options).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'password'" do
+        let(:ask) { base_ask.merge("password" => true) }
+
+        it "creates two Password widgets" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:Password).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], ask["default"]).
+            and_call_original
+          expect(client).to receive(:Password).
+            with(client.Id(:pass2), client.Opt(:notify), "", ask["default"]).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'static_text'" do
+        let(:ask) { base_ask.merge("type" => "static_text") }
+
+        it "creates a Label widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:Label).
+            with(client.Id("0_0"), ask["default"]).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains question with type 'symbol'" do
+        let(:ask) {
+          base_ask.merge("type" => "symbol", "default" => :desktop, "selection" => items)
+        }
+        let(:items) {
+          %w(desktop server).map { |i| { "value" => i.to_sym, "label" => i.capitalize } }
+        }
+
+        it "creates a ComboBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expected_options = [
+            client.Item(client.Id(:desktop), "Desktop", true),
+            client.Item(client.Id(:server), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], expected_options).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'boolean'" do
+        let(:ask) do
+          base_ask.merge("type" => "boolean", "question" => "Register system?", "default" => "true")
+        end
+
+        it "creates a CheckBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:CheckBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], true).
```

Comment:
maybe it would be nice to have test with more then one widget, to see if id is properly created.